data/scenarios/Testing/479-atomic-race.yaml

```diff
@@ -0,0 +1,76 @@
+# This scenario is specially engineered to almost guarantee a crash
+# due to a race condition in short order.
+#
+#   - The steps per tick is set to 1, so robots are interrupted after
+#     every single step, making bugs due to bad interleaving much more
+#     probable.
+#   - A `weed` entity is used which grows very quickly
+#   - `harvest` is used, which causes more weeds to grow in the same spot
+#   - A crash happens when the planter robot tests for the presence of a weed
+#     and doesn't find one, but then is interrupted and a new weed grows;
+#     then the planter robot crashes when it tries to place a weed.
+#   - The harvesting robot waits a random number of ticks between 1-4 each
+#     time, to give the planting robot time to run into the race condition.
+#
+# This may work slightly less well once we implement #502, but I think it should
+# still work.
+
+name: Grab/harvest/place (with race)
+description: |
+  Make sure we can cause a robot to crash due to a race condition, in order to
+  know that atomic blocks are doing their job.
+  https://github.com/swarm-game/swarm/issues/479
+stepsPerTick: 1
+creative: true
+win: |
+  try {
+    p <- robotNamed "planter";
+    as p {n <- count "weed"; return (n == 0)}
+  } { return false }
```

Comment:
I find testing this later with `expectedFail` confusing since we otherwise use it for bugs that need fixing.

I would suggest winning by creating the Win entity when harvesting fails.

Something like `s/tryharvest/try {raceharvest} {create "Win"}/g`.

Reply:
Oh, good point, and good suggestion!  It's actually the planter robot that is expected to crash but yes, a `try {...} {create "Win"}` does the trick just fine.